cdap-cli/src/main/java/co/cask/cdap/cli/command/CallServiceCommand.java

```diff
@@ -87,7 +89,17 @@ public void perform(Arguments arguments, PrintStream output) throws Exception {
     }
 
     Map<String, String> headerMap = GSON.fromJson(headers, new TypeToken<Map<String, String>>() { }.getType());
-    URL url = new URL(serviceClient.getServiceURL(service), path);
+    String[] idParts = parseProgramIdArgument(arguments, ArgumentName.SERVICE.getName());
+    URL url;
+    if (idParts[VERSION_IDX] == null) {
```

Comment:
instead, when ``idParts[VERSION_IDX]`` is ``null``, why can't we use default version? That way, you don't need this if-else, right?